libs/blocks/merch/merch.js

```diff
@@ -122,6 +122,34 @@ const LOADING_ENTITLEMENTS = 'loading-entitlements';
 let log;
 let upgradeOffer = null;
 
+/**
+ * Given a url, calculates the hostname of MAS platform.
+ * Supports, www prod, stage, local and feature branches.
+ * @param {string} hostname
+ * @param {string} maslibs
+ * @returns base url for mas platform
+ */
+export function getMasBase(hostname, maslibs) {
+  let { baseUrl } = getMasBase;
+  if (baseUrl) return baseUrl;
+  baseUrl = 'https://www.adobe.com/mas';
+  if (!maslibs) return baseUrl;
+  if (maslibs === 'stage') {
+    baseUrl = 'https://www.stage.adobe.com/mas';
+  } else if (maslibs === 'local') {
+    baseUrl = 'http://localhost:9001';
+  } else {
+    const extension = /.live$/.test(hostname) ? 'live' : 'page';
+    if (/--/.test(maslibs)) {
+      baseUrl = `https://${maslibs}.hlx.${extension}`;
+    } else {
+      baseUrl = `https://${maslibs}--mas--adobecom.hlx.${extension}`;
```

Comment:
how about we make 'maslibs' work the same way as 'milolibs' do?

`?maslibs=mwpw-153962--mas--yesil`
branch-fork-owner
Then you can save precious bytes and remove stage and prod cases, since the link could be given as
`?maslibs=main--mas--adobecom` or `?maslibs=stage--mas--adobecom` (when/if we have stage) instead of `?maslibs=stage`

I know the latter is shorter but it involves an additional learning curve for devs and IMO is not worth the space it takes.

Reply:
During stage testing, I think  just using `&maslibs=stage` instead of `&maslibs=stage--mas--adobecom` will save a lot of time to humans. 
But saving bytes is also interesting. @npeltier @narcis-radu @3ch023 @overmyheadandbody 
👍 or 👎  for saving bytes?

Reply:
i think i prefer mariia/milolibs's approach, yes :)

Reply:
fine

Reply:
updated